Platon/backend/app/workspace_system/views.py

```diff
@@ -0,0 +1,154 @@
+    ),
+})
+
+trending_workspaces_model = api.model('Trending Projects', {
+    'trending_projects': fields.List(
+        fields.Nested(workspace_small_model)
+    )
+})
+
+@workspace_system_ns.route("/self")
+class GetSelfWorkspaces(Resource):
+
+    @api.doc(responses={401:'Authantication Problem',500:'Database Connection Problem'})
+    @api.response(200,'Valid Response',workspace_list_model)
+    @api.expect(get_self_parser)
+    @login_required
+    def get(user_id,self):
+        try:
+            contributions = Contribution.query.filter(Contribution.user_id==user_id).all()
+        except:
+            return make_response(jsonify({"err": "Database Connection Error"}),500)
+        workspaces = []
+        for contribution in contributions:
+            if contribution.is_active == 0:
+                continue
+            try:
+                ws = Workspace.query.get(contribution.workspace_id)
+            except:
+                return make_response(jsonify({"err": "Database Connection Error"}),500)
+            try:
+                contributors = Contribution.query.filter(Contribution.workspace_id == ws.id).all()
+            except:
+                return make_response(jsonify({"err": "Database Connection Error"}),500)
+            contributor_list = []
+            for contributor in contributors:
+                try:
+                    user = User.query.get(contributor.user_id)
+                except:
+                    return make_response(jsonify({"err": "Database Connection Error"}),500)
+                contributor_list.append({
+                        "id": user.id,
+                        "name": user.name,
+                        "surname": user.surname
+                })
+            ws_dict = {
+                "id": ws.id,
+                "is_private": int(ws.is_private),
+                "title": ws.title,
+                "state": ws.state,
+                "creation_time": ws.timestamp,
+                "description": ws.description,
+                "deadline": ws.deadline,
+                "max_collaborators": ws.max_collaborators,
+                "contributors": contributor_list
+            }
+            workspaces.append(ws_dict)
+        return make_response(jsonify({"workspaces": workspaces}),200)
+
+@workspace_system_ns.route("/trending_projects")
+class TrendingWorkspacesAPI(Resource):
+
+    @api.doc(responses={500 : 'Database Connection Error',400: 'Invalid Input Format'})
+    @api.response(200, 'Valid Response', trending_workspaces_model)
+    @api.expect(trending_project_parser)
+    def get(self):
+        form = TrendingProjectsForm(request.args)
+        if form.validate():
+            try:
+                all_workspaces = Workspace.query.order_by(Workspace.trending_score.desc()).all()
+            except:
+                return make_response(jsonify({"err": "Database Connection Error"}),500)
+            all_workspaces = all_workspaces[:form.number_of_workspaces.data]
```

Comment:
A very minor suggestion: in the future, we can move these parts regarding information retrieval   from the database to helpers for a more brief code. (But it is not necessary at the moment.)